test/libyul/YulOptimizerTest.cpp

```diff
@@ -352,21 +350,34 @@ TestCase::TestResult YulOptimizerTest::run(ostream& _stream, string const& _line
 		return TestResult::FatalError;
 	}
 
-	m_obtainedResult = AsmPrinter{*m_dialect}(*m_ast) + "\n";
 
-	if (m_optimizerStep != m_step)
+	string const step = "step:";
+	bool hasStep = boost::algorithm::starts_with(m_expectation, step);
+	if (!hasStep)
+	{
+		AnsiColorized(_stream, _formatted, { formatting::BOLD, formatting::RED }) << _linePrefix << "Expected optimizer step not found." << endl;
+		return TestResult::FatalError;
```

Comment:
Is this really a fatal error? Shouldn't it just be an expectation mismatch?

Reply:
Then you would also not need to remove it from `m_expectation' in the following lines.